apps/code-examples/src/app/code-examples/ag-grid/data-grid/data-manager-multiselect/data-manager-multiselect-data-grid-docs-demo-view-grid.component.ts

```diff
@@ -0,0 +1,302 @@
+    this.displayedItems = this.items;
+
+    this.dataManagerService.initDataView(this.viewConfig);
+
+    this.gridOptions = this.skyAgGridService.getGridOptions({
+      gridOptions: {
+        columnDefs: this.multiSelectColumnDefinitions,
+        rowSelection: 'multiple',
+        onGridReady: this.onGridReady.bind(this),
+      },
+    });
+
+    this.dataManagerService
+      .getDataStateUpdates(this.viewId)
+      .subscribe((state) => {
+        this.dataState = state;
+        this.updateColumnOrder();
+        this.isGridReadyForInitialization = true;
+        this.updateDisplayedItems();
+        this.changeDetector.detectChanges();
+      });
+
+    this.dataManagerService.getActiveViewIdUpdates().subscribe((id) => {
+      this.isThisViewActive = id === this.viewId;
+      this.changeDetector.detectChanges();
+    });
+  }
+
+  public onGridReady(gridReadyEvent: GridReadyEvent): void {
+    this.gridApi = gridReadyEvent.api;
+    this.gridApi.sizeColumnsToFit();
+    this.columnApi = gridReadyEvent.columnApi;
+    this.updateDisplayedItems();
+    this.changeDetector.markForCheck();
+  }
+
+  public onRowSelected(rowSelectedEvent: RowSelectedEvent): void {
+    if (!rowSelectedEvent.data.selected) {
+      this.updateDisplayedItems();
+    }
+  }
+
+  private searchItems(items: any[]): any[] {
+    let searchedItems = items;
+    const searchText = this.dataState && this.dataState.searchText;
+
+    if (searchText) {
+      searchedItems = items.filter(function (item: any) {
+        let property: any;
+
+        for (property in item) {
+          if (
+            Object.prototype.hasOwnProperty.call(item, property) &&
+            (property === 'name' || property === 'description')
+          ) {
+            const propertyText = item[property].toLowerCase();
+            if (propertyText.indexOf(searchText) > -1) {
+              return true;
+            }
+          }
+        }
+        return false;
+      });
+    }
+    return searchedItems;
+  }
+
+  private filterItems(items: any[]): any[] {
+    let filteredItems = items;
+    const filterData = this.dataState && this.dataState.filterData;
+
+    if (filterData && filterData.filters) {
+      const filters = filterData.filters;
+      filteredItems = items.filter((item: any) => {
+        return (
+          ((filters.hideSales && item.department.name !== 'Sales') ||
+            !filters.hideSales) &&
+          ((filters.jobTitle !== 'any' &&
+            item.jobTitle?.name === filters.jobTitle) ||
+            !filters.jobTitle ||
+            filters.jobTitle === 'any')
+        );
+      });
+    }
+
+    return filteredItems;
+  }
+
+  private endDateFormatter(params: ValueFormatterParams): string {
+    const dateConfig = { year: 'numeric', month: '2-digit', day: '2-digit' };
+    return params.value
+      ? params.value.toLocaleDateString('en-us', dateConfig)
+      : 'N/A';
+  }
+
+  // TODO: put this logic in the skyAgGridDataManagerAdapter
```

Comment:
lets make a ticket for this so we can keep TODOs out of the code consumers will see

Reply:
done in: [remove TODOs that are covered by PM](https://github.com/blackbaud/skyux/pull/531/commits/8f70909f772bc613713865d16ae2af19d08d7012)